test/unit/cross_lane_ops_test/device/dpp_ops.hpp

```diff
@@ -0,0 +1,306 @@
+    ROCWMMA_DEVICE inline uint32_t getDppShiftExpect(uint32_t input, uint32_t prev)
+    {
+        input += ShiftDir == CrossLaneOps::OP_DIR_L ? ShiftDist : -ShiftDist;
+        uint32_t fillValue  = BoundCtrl ? 0 : prev;
+        auto     afterShift = (input & (SubGroupSize - 1));
+        afterShift += ShiftDir == CrossLaneOps::OP_DIR_L ? -ShiftDist : ShiftDist;
+        return (afterShift < 0 || afterShift >= SubGroupSize) ? fillValue : input;
+    }
+
+    template <uint32_t Select0, uint32_t Select1>
+    ROCWMMA_DEVICE inline uint32_t getDppShuffle2Expect(uint32_t input)
+    {
+        auto id = input & 0b1;
+        input -= id;
+        input += id == 0 ? Select0 : Select1;
+        return input;
+    }
+
+    template <uint32_t Select0, uint32_t Select1, uint32_t Select2, uint32_t Select3>
+    ROCWMMA_DEVICE inline uint32_t getDppShuffle4Expect(uint32_t input)
+    {
+        auto id = input & 0b11;
+        input -= id;
+        input += id == 0 ? Select0 : id == 1 ? Select1 : id == 2 ? Select2 : Select3;
+        return input;
+    }
+
+    template <uint32_t SubGroupSize>
+    ROCWMMA_DEVICE inline uint32_t getDppSwapExpect(uint32_t input)
+    {
+        return input ^ SubGroupSize;
+    }
+
+    template <uint32_t SubGroupSize>
+    ROCWMMA_DEVICE inline uint32_t getDppWFallBCastExpect(uint32_t input)
+    {
+        if constexpr(SubGroupSize == 16)
+        {
+            auto firstInRow = input & 0b110000;
+            input           = firstInRow > 0 ? firstInRow - 1 : input;
+        }
+        else
+        {
+            auto firstInRow = input & 0b100000;
+            input           = firstInRow > 0 ? firstInRow - 1 : input;
+        }
+        return input;
+    }
+
+    template <typename DataT,
+              typename CrossLaneOp,
+              uint32_t WriteRowMask,
+              uint32_t WriteBankMask,
+              bool     BoundCtrl>
+    ROCWMMA_DEVICE std::enable_if_t<CrossLaneOp::opId() == CrossLaneOps::OP_ID_BCAST
+                                        && CrossLaneOp::opImpl() == CrossLaneOps::OP_IMPL_DPP,
+                                    bool>
+                   dppOpsTestCase()
+    {
+        uint32_t id       = threadIdx.x;
+        uint32_t prev     = VALUE_OUT_OF_RANGE;
+        DataT    input    = makeValueFromU32<DataT>(id);
+        bool     isMasked = isDppMasked(id, WriteRowMask, WriteBankMask);
+        DataT    expect   = makeValueFromU32<DataT>(
+            isMasked ? getDppBCastExpect<CrossLaneOp::GROUP_SIZE, CrossLaneOp::ELEMENT_IDX>(id)
+                          : prev);
+
+        auto output
+            = rocwmma::Dpp::Driver<CrossLaneOp, WriteRowMask, WriteBankMask, BoundCtrl>::exec(
+                input, makeValueFromU32<DataT>(prev));
+
+        // printf("op 0, input %d, WriteRowMask %x, WriteBankMask %x, BoundCtrl %d, expect %d, output %d\n",  input , WriteRowMask , WriteBankMask , BoundCtrl, expect , output );
```

Comment:
Suggestion: remove the commented out code, or put into #if !NDEBUG ...#endif guards

Reply:
Removed all printf. 

Removed them in the last commit in case I need them in the review loop.